sqlglot/dialects/tsql.py

```diff
@@ -87,6 +89,29 @@ def generate_format_sql(self, e):
     return f"FORMAT({self.format_args(e.this, fmt)})"
 
 
+def _string_agg_sql(self, e):
+    e = e.copy()
+
+    this = e.this
+    distinct = e.find(exp.Distinct)
+    if distinct:
+        # exp.Distinct can appear below an exp.Order or an exp.GroupConcat expression
+        self.unsupported("T-SQL STRING_AGG doesn't support DISTINCT.")
+        this = distinct.expressions[0]
+        distinct.pop()
+
+    order = ""
+    if isinstance(e.this, exp.Order):
+        if e.this.this:
+            this = e.this.this
+            e.this.this.pop()
+        order = f" WITHIN GROUP ({self.sql(e.this)[1:]})"
+
+    separator = e.args.get("separator")
+    separator = separator.name if separator else exp.Literal.string(",")
```

Comment:
why are using Literal.string? why not just ","

Reply:
Because the formatting wasn't quite right. If we use a raw string it outputs `STRING_AGG(x, ,)` for example.